src/modules/cpu/kernel/warp_perspective.hpp

```diff
@@ -0,0 +1,2286 @@
+    locW = std::fma(dstX, perspectiveMatrix_f9->data[6], std::fma(dstY, perspectiveMatrix_f9->data[7], perspectiveMatrix_f9->data[8]));
+}
+
+inline void compute_warp_perspective_src_loc(Rpp32s dstY, Rpp32s dstX, Rpp32f &locW, Rpp32f &locY, Rpp32f &locX, Rpp32f &srcY, Rpp32f &srcX, Rpp32f9 *perspectiveMatrix_f9, Rpp32s roiHalfHeight, Rpp32s roiHalfWidth)
+{
+    dstX -= roiHalfWidth;
+    dstY -= roiHalfHeight;
+    locX = std::fma(dstX, perspectiveMatrix_f9->data[0], std::fma(dstY, perspectiveMatrix_f9->data[1], perspectiveMatrix_f9->data[2]));
+    locY = std::fma(dstX, perspectiveMatrix_f9->data[3], std::fma(dstY, perspectiveMatrix_f9->data[4], perspectiveMatrix_f9->data[5]));
+    locW = std::fma(dstX, perspectiveMatrix_f9->data[6], std::fma(dstY, perspectiveMatrix_f9->data[7], perspectiveMatrix_f9->data[8]));
+    srcX = (locX/locW) + roiHalfWidth;
+    srcY = (locY/locW) + roiHalfHeight;
+}
+
+inline void compute_warp_perspective_src_loc_next_term(Rpp32s dstX, Rpp32f &locW, Rpp32f &locY, Rpp32f &locX, Rpp32f &srcY, Rpp32f &srcX, Rpp32f9 *perspectiveMatrix_f9, Rpp32s roiHalfHeight, Rpp32s roiHalfWidth)
+{
+    locW += perspectiveMatrix_f9->data[6];
+    locY += perspectiveMatrix_f9->data[3];   // Used in computation of next srcY locations by adding the delta from previous location
+    locX += perspectiveMatrix_f9->data[0];   // Used in computation of next srcX locations by adding the delta from previous location
+    srcX = (locX/locW) + roiHalfWidth;
+    srcY = (locY/locW) + roiHalfHeight;
+}
+
+/************* NEAREST NEIGHBOR INTERPOLATION *************/
+
+RppStatus warp_perspective_nn_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                           RpptDescPtr srcDescPtr,
+                                           Rpp8u *dstPtr,
+                                           RpptDescPtr dstDescPtr,
+                                           Rpp32f *perspectiveTensor,
+                                           RpptROIPtr roiTensorPtrSrc,
+                                           RpptRoiType roiType,
+                                           RppLayoutParams srcLayoutParams,
+                                           rpp::Handle& handle)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi, roiLTRB;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+        compute_ltrb_from_xywh_host(&roi, &roiLTRB);
+        Rpp32s roiHalfWidth = roi.xywhROI.roiWidth >> 1;
+        Rpp32s roiHalfHeight = roi.xywhROI.roiHeight >> 1;
+
+        Rpp32f9 *perspectiveMatrix_f9;
+        perspectiveMatrix_f9 = (Rpp32f9 *)perspectiveTensor + batchCount;
```

Comment:
Static cast doesn't work here, Used reinterpret cast
Also used perspectiveTensor + batchCount * 9 instead of the suggestion because the change then points to a float value which cannot be casted to a pointer